### PR TITLE
H183-cleanup: make per-channel surface heads default, remove flag

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

PR #1510 (H183) merged per-channel surface decoder heads as the new single-model SOTA (test_WSS=6.4427%). The merged code added `--per-channel-surface-heads` as an opt-in CLI flag and kept the old shared-MLP path as the default for backward compatibility.

Now that per-channel heads is the confirmed SOTA, the shared-head path is dead code. All future experiments should always use per-channel heads. This cleanup removes the flag, makes per-channel heads unconditional, and deletes the legacy shared-MLP path.

**This is a code-quality cleanup, NOT a new experiment.** No long training run required — just a 1-EP smoke to verify correctness.

### Changes to make

**In `target/model.py`:**
1. Remove the `per_channel_surface_heads` parameter from the `Transolver.__init__` signature
2. Remove the `if per_channel_surface_heads:` / `else:` branch — keep ONLY the `nn.ModuleList` per-channel path (the `if` branch). Delete the shared MLP `else` branch entirely.
3. Remove the `self.surface_out = None` dead-code alias
4. The result: `self.surface_out_heads` is always created, no condition needed

**In `target/train.py`:**
1. Remove the `--per-channel-surface-heads` CLI argument (or its equivalent in the Config class)
2. Remove passing `per_channel_surface_heads=...` to the model constructor
3. The result: the flag no longer exists; per-channel heads are always on

### Smoke run (1 EP DDP8)

After making the code changes, run a 1-EP smoke to confirm:
- Training starts without errors
- val metrics are finite
- The `--per-channel-surface-heads` flag no longer exists in the CLI help

```bash
torchrun --nproc_per_node=8 train.py \
  --data-root /mnt/new-pvc/Processed/drivaerml_processed_rawcanon_20260511 \
  --optimizer lion --lr 1e-4 --weight-decay 0.005 --lion-beta1 0.95 --lion-beta2 0.98 \
  --batch-size 1 --lr-warmup-epochs 1 --lr-cosine-t-max 1 --epochs 1 \
  --model-pe string_multisigma --pe-init-sigmas 0.25,0.5,1.0,2.0,4.0 --pe-num-features 16 \
  --model-layers 6 --model-hidden-dim 512 --model-heads 4 --model-slices 128 \
  --vol-p-charbonnier-weight 0.1 --wss-charbonnier-weight 0.1 --wss-charbonnier-axes z \
  --use-gradnorm --gradnorm-alpha 0.5 --gradnorm-min-w-vol-p 0.15 \
  --use-curvature-attention-bias --use-y-symmetry-aug --y-symmetry-aug-prob 0.5 \
  --surface-out-width-factor 2.0 \
  --train-surface-points 65000 --train-volume-points 65000 \
  --eval-surface-points 65536 --eval-volume-points 65536 \
  --ema-decay 0.999 --ema-start-step 500 \
  --agent dl24-tanjiro \
  --wandb-group h183-cleanup \
  --wandb-name dl24-tanjiro/h183-cleanup-smoke
```

**Smoke gates:**
- `val_WSS ≤ 13.5%` (H183 EP1 was 12.973%)
- `nonfinite_grads = 0`
- Confirm `--per-channel-surface-heads` flag no longer in CLI help output

### Post smoke results

Post EP1 val metrics as a comment, confirm the flag is gone, then mark PR ready for review.

## Baseline (H183 SOTA — PR #1510, run `guw83mge`)

| Metric | H183 SOTA | Floor |
|---|---:|---:|
| test_WSS | **6.4427%** ⭐ | PRIMARY |
| test_SP | 3.5187% | ≤ 3.577% |
| test_VP | 3.4415% | ≤ 3.643% |
| test_ABU | 5.6152% | ≤ 5.844% |
| test_τ_x | 5.6983% | track |
| test_τ_y | 6.9813% | track |
| test_τ_z | 8.4364% | track |

H183 reproduce command (this is now the default — no `--per-channel-surface-heads` needed after this cleanup):
```bash
torchrun --nproc_per_node=8 train.py \
  --data-root /mnt/new-pvc/Processed/drivaerml_processed_rawcanon_20260511 \
  --optimizer lion --lr 1e-4 --weight-decay 0.005 --lion-beta1 0.95 --lion-beta2 0.98 \
  --batch-size 1 --lr-warmup-epochs 1 --lr-cosine-t-max 30 --epochs 30 \
  --model-pe string_multisigma --pe-init-sigmas 0.25,0.5,1.0,2.0,4.0 --pe-num-features 16 \
  --model-layers 6 --model-hidden-dim 512 --model-heads 4 --model-slices 128 \
  --vol-p-charbonnier-weight 0.1 --wss-charbonnier-weight 0.1 --wss-charbonnier-axes z \
  --use-gradnorm --gradnorm-alpha 0.5 --gradnorm-min-w-vol-p 0.15 \
  --use-curvature-attention-bias --use-y-symmetry-aug --y-symmetry-aug-prob 0.5 \
  --surface-out-width-factor 2.0 \
  --train-surface-points 65000 --train-volume-points 65000 \
  --eval-surface-points 65536 --eval-volume-points 65536 \
  --ema-decay 0.999 --ema-start-step 500 \
  --agent dl24-tanjiro
```

## Terminal result format

After smoke passes and PR is ready for review, post:
```
SENPAI-RESULT: {"terminal":true,"status":"complete","pending_arms":false,"wandb_run_ids":["<smoke-run-id>"],"primary_metric":{"name":"val_primary/abupt_axis_mean_rel_l2_pct","value":<ep1_val>},"test_metric":{"name":"test_primary/wall_shear_rel_l2_pct","value":0.0}}
```

### CODE REVIEW

**Comment by morganmcg1** [comment]
STUDENT dl24-tanjiro:
SENPAI-RESULT: {"terminal":true,"status":"complete","pending_arms":false,"wandb_run_ids":["7z38enbq"],"primary_metric":{"name":"val_primary/abupt_axis_mean_rel_l2_pct","value":12.785693742392828},"test_metric":{"name":"test_primary/wall_shear_rel_l2_pct","value":0.0}}

## Results

Code cleanup PR — per-channel surface heads are now the unconditional default; `--per-channel-surface-heads` flag is fully removed. 1-EP DDP8 smoke confirms no regression vs the H183 SOTA EP1 baseline.

### Smoke gates

| Gate | Target | Observed | Status |
|---|---:|---:|:---:|
| `val_WSS` (`val_primary/wall_shear_rel_l2_pct`) | ≤ 13.5% | **12.691%** | PASS (slightly better than H183 EP1's 12.973%) |
| `nonfinite_grads` | 0 | 0 across **all** modules (`train/grad/nonfinite_count = 0`, every per-module `nonfinite_count = 0`) | PASS |
| `--per-channel-surface-heads` flag in CLI | absent | absent (`python train.py --help` has zero matches for `per[-_]channel`) | PASS |

### EP1 validation metrics (rank0 run `7z38enbq`)

| Metric | Value (%) |
|---|---:|
| `val_primary/abupt_axis_mean_rel_l2_pct` (PRIMARY) | **12.786** |
| `val_primary/wall_shear_rel_l2_pct` | 12.691 |
| `val_primary/surface_pressure_rel_l2_pct` | 8.609 |
| `val_primary/volume_pressure_rel_l2_pct` | 13.649 |
| `val_primary/wall_shear_x_rel_l2_pct` | 11.308 |
| `val_primary/wall_shear_y_rel_l2_pct` | 13.958 |
| `val_primary/wall_shear_z_rel_l2_pct` | 16.404 |

### Comparison vs H183 EP1 baseline

H183 EP1 reported `val_WSS=12.973%`. This cleanup smoke: `val_WSS=12.691%`. Within expected EP1 variance from data shuffling/y-symmetry-aug RNG — model wiring is byte-equivalent to the H183 default path.

### Code changes

**`model.py` `SurfaceTransolver`:**
- Removed `per_channel_surface_heads: bool = False` parameter from `__init__` signature.
- Removed `self.per_channel_surface_heads` attribute.
- Deleted the `if self.per_channel_surface_heads: ... else: ...` branch; only the per-channel `nn.ModuleList` path remains.
- Deleted the `self.surface_out = None` / `self.surface_out_heads = None` dead-code aliases — `self.surface_out_heads` is unconditionally constructed.
- Simplified the forward pass: removed the `if self.surface_out_heads is not None:` guard; the per-channel concatenation is now the only path.
- Consolidated the comment block above the head construction to reflect H183 as the active design.

**`train.py`:**
- Removed `per_channel_surface_heads: bool = False` from the `Config` dataclass.
- Removed the corresponding entry from the CLI help/description dict.
- Removed `per_channel_surface_heads=config.per_channel_surface_heads` from the `build_model` call.

Net diff: ~30 LOC removed, no LOC added. One obvious training path remains.

### Exact command used

```bash
torchrun --nproc_per_node=8 train.py \
  --data-root /mnt/new-pvc/Processed/drivaerml_processed_rawcanon_20260511 \
  --optimizer lion --lr 1e-4 --weight-decay 0.005 --lion-beta1 0.95 --lion-beta2 0.98 \
  --batch-size 1 --lr-warmup-epochs 1 --lr-cosine-t-max 1 --epochs 1 \
  --model-pe string_multisigma --pe-init-sigmas 0.25,0.5,1.0,2.0,4.0 --pe-num-features 16 \
  --model-layers 6 --model-hidden-dim 512 --model-heads 4 --model-slices 128 \
  --vol-p-charbonnier-weight 0.1 --wss-charbonnier-weight 0.1 --wss-charbonnier-axes z \
  --use-gradnorm --gradnorm-alpha 0.5 --gradnorm-min-w-vol-p 0.15 \
  --use-curvature-attention-bias --use-y-symmetry-aug --y-symmetry-aug-prob 0.5 \
  --surface-out-width-factor 2.0 \
  --train-surface-points 65000 --train-volume-points 65000 \
  --eval-surface-points 65536 --eval-volume-points 65536 \
  --ema-decay 0.999 --ema-start-step 500 \
  --agent dl24-tanjiro \
  --wandb-group h183-cleanup \
  --wandb-name dl24-tanjiro/h183-cleanup-smoke
```

Training time: 50.4 min wall-clock (EP1, 10975 train steps).

### Peak memory

Peak GPU memory ~**51.5 GB** (≈50% of H100 97 GB) across ranks. No OOM, no near-limit signals. Identical envelope to H183.

### W&B run ID

Primary rank0: [`7z38enbq`](https://wandb.ai/wandb-applied-ai-team/senpai-v1-drivaerml-ddp8/runs/7z38enbq) (run name `dl24-tanjiro/h183-cleanup-smoke-rank0`, group `h183-cleanup`)

### What happened

- Code changes implement the cleanup exactly as specified in the PR instructions: remove the boolean flag, remove the shared-MLP branch, make `surface_out_heads` unconditional.
- 1-EP smoke trained cleanly to completion. Val metrics finite, no NaNs/Infs in any gradient module, and `val_WSS` lands within reasonable EP1 RNG variance of H183's EP1 (12.691 vs 12.973). This is the expected behavior — the cleanup is a no-op on the wire; only dead code was removed.
- The `--per-channel-surface-heads` flag is verifiably gone from the CLI surface area.

### What was NOT touched

- `BASELINE.md` and `research/RESEARCH_IDEAS_*.md` still reference `--per-channel-surface-heads` as historical context. These are intentionally left as historical records; the PR scope is code-only.

### Suggested follow-ups

- None needed for this PR. Future PRs should drop the `--per-channel-surface-heads` flag from their reproduce commands (the H183 reproduce in the PR body already does so).
- (Out of scope, just flagging) If the next stack-cleanup pass wants to remove other now-dead historical flags, the same pattern applies — but each such removal warrants its own PR + smoke.

**Comment by morganmcg1** [comment]
ADVISOR 05:00Z — MERGING ✅

Clean code cleanup with all smoke gates PASSED:
- val_WSS=12.691% (within EP1 RNG variance of H183 EP1=12.973%) ✓
- nonfinite_grads=0 across all modules ✓
- `--per-channel-surface-heads` flag confirmed absent from CLI help ✓
- ~30 LOC removed, no LOC added — H183 stack is now the unconditional default ✓

Per-channel surface heads are now the single training path. All subsequent experiments inherit this default — no flag needed in reproduce commands. This unblocks H188/H189 from carrying a stale-default workaround.

Squash-merging now. Tanjiro — next assignment incoming after merge completes.
